src/flagsView.ts

```diff
@@ -0,0 +1,452 @@
+import * as vscode from 'vscode';
+import { FeatureFlag } from './models';
+import { LaunchDarklyAPI } from './api';
+import { Configuration, getIsTreeviewEnabled } from './configuration';
+import { FlagStore } from './flagStore';
+import * as path from 'path';
+
+export class ldFeatureFlagsProvider implements vscode.TreeDataProvider<FlagValue> {
+	private readonly api: LaunchDarklyAPI;
+	private config: Configuration;
+	private flagStore: FlagStore;
+	private flagValues: Array<FlagValue>;
+	private ctx: vscode.ExtensionContext;
+	private _onDidChangeTreeData: vscode.EventEmitter<FlagValue | undefined> = new vscode.EventEmitter<
+		FlagValue | undefined
+	>();
+	readonly onDidChangeTreeData: vscode.Event<FlagValue | undefined> = this._onDidChangeTreeData.event;
+
+	constructor(api: LaunchDarklyAPI, config: Configuration, flagStore: FlagStore, ctx: vscode.ExtensionContext) {
+		this.api = api;
+		this.config = config;
+		this.ctx = ctx;
+		this.flagStore = flagStore;
+		this.start();
+	}
+
+	refresh(): void {
+		this._onDidChangeTreeData.fire();
+	}
+
+	reload(): void {
+		this.getFlags();
+		this.refresh();
+	}
+
+	getTreeItem(element: FlagValue): vscode.TreeItem {
+		return element;
+	}
+
+	getChildren(element?: FlagValue): Thenable<FlagValue[]> {
+		if (!this.flagValues) {
+			return Promise.resolve([new FlagValue(this.ctx, 'No Flags Found.', vscode.TreeItemCollapsibleState.None)]);
+		}
+
+		if (element) {
+			return Promise.resolve(element.children);
+		} else {
+			return Promise.resolve(
+				this.flagValues.map(function(flag) {
+					return flag;
+				}),
+			);
+		}
+	}
+
+	async getFlags() {
+		const flags = await this.api.getFeatureFlags(this.config.project, this.config.env);
+		let flagValues = [];
+		for (const flag of flags) {
+			let item = this.flagToValues(flag);
+			flagValues.push(item);
+		}
+		this.flagValues = flagValues;
+		this.refresh();
+	}
+
+	async start() {
+		this.ctx.subscriptions.push(
+			vscode.commands.registerCommand('ldFeatureFlags.copyKey', (node: FlagValue) =>
+				vscode.env.clipboard.writeText(node.label.split(':')[1].trim()),
+			),
+			vscode.commands.registerCommand('ldFeatureFlags.openBrowser', (node: FlagValue) =>
+				vscode.env.openExternal(vscode.Uri.parse(node.uri)),
+			),
```

Comment:
Might have coverted this in another comment, but maybe we can just make these internal behavior instead of an exposed command?

Reply:
I'm not sure what you mean here with Internal behavior instead of an exposed command? This is what's needed for the button to open in the browser with the treeview.

Reply:
Ah got it, I wasn't completely sure if the button in the tree view needed to be tied to a command - since it is, sounds like we should keep this command

Reply:
Although I wonder if there's any way to remove the extra commands that show up in the command palette: https://github.com/launchdarkly/ld-vscode/pull/39#discussion_r420991541

Reply:
I was able to hide them!